### PR TITLE
setup for jest/eslint + workflows

### LINKED ISSUE #5: Setup jest e eslint

### PR DESCRIPTION
**Pull request by Willian1661**
close #5

### CODE REVIEW

File: client/interfaces/post.ts [hunk 1]
```diff
@@ -1,6 +1,6 @@
-import type Author from './author'
+import type Author from './author';
 
-type PostType = {
+interface PostType {
```
Comment (eerison) [inline client/interfaces/post.ts:3]: Could you just make sure if the application is still running?

File: client/package.json [hunk 2]
```diff
@@ -2,9 +2,12 @@
   "private": true,
   "scripts": {
     "dev": "next",
+    "test": "jest --coverage",
+    "lint": "eslint --ext .js,.ts .",
+    "lint:fix": "npm run lint -- --fix",
     "build": "next build",
     "start": "next start",
-    "typecheck": "tsc"
+    "typecheck": "tsc --watch"
```
Comment (eerison) [inline client/package.json]: create a new entry for watch

"typecheck:watch"

**Review by eerison — CHANGES_REQUESTED** [review]

File: client/interfaces/post.ts (hunk 1 above)
Comment (Willian1661) [reply client/interfaces/post.ts:3]: ok

Comment (Willian1661) [reply client/interfaces/post.ts:3]: yea it's still runnig

File: client/package.json (hunk 2 above)
Comment (Willian1661) [reply client/package.json]: this typecheck is just for call ts package ?

Comment (eerison) [reply client/package.json]: I don't know what this `tsc` does, But if you want to add this parameter, then create a new entry

Comment (Willian1661) [reply client/package.json]: i saw that it returns the typescript version i guess,i thiink it's nothing big

Comment (eerison) [reply client/package.json]: but no make sense you add `--watch` in `typecheck`

Comment (Willian1661) [reply client/package.json]: `npx tsc.`

"You can include the above command in your scripts to make it easier to run. Go to package.json and add the` --watch` flag to watch the files for changes. "

i think this is to update the changes of the page's build,
i ran "typecheck" and it returns the build info.

Comment (Willian1661) [reply client/package.json]: i removed it

**Review by eerison — APPROVED** [review]

**Pull request merged**